### PR TITLE
feat: implement `HTMX` plugin using `litestar-htmx`

### PR DESCRIPTION
**Pull request by cofin**
## Description

This plugin migrates the HTMX integration to `litestar.plugins.htmx`.  

This logic has been moved to it's own repository named `litestar-htmx`

This is in preparation for 3.0 release where `contrib` will be deprecated.

### CODE REVIEW

File: litestar/contrib/htmx/_utils.py [hunk 1]
```diff
@@ -1,12 +1,22 @@
...
-from typing import TYPE_CHECKING, Any, Callable, cast
-from urllib.parse import quote
+from typing import TYPE_CHECKING
 
-from litestar.exceptions import ImproperlyConfiguredException
-from litestar.serialization import encode_json
+from litestar.utils import warn_deprecation
 
+if TYPE_CHECKING:
+    from litestar_htmx._utils import (  # noqa: TCH004
```
Comment (Alc-Alc) [inline litestar/contrib/htmx/_utils.py:8]: While this is something we control, is it good practice to import a third party "private" module inside Litestar's private module?

File: pyproject.toml [hunk 2]
```diff
@@ -44,6 +44,8 @@ dependencies = [
     "click",
     "rich>=13.0.0",
     "rich-click",
+    # default litestar plugins
```
Comment (Alc-Alc) [inline pyproject.toml:47]: Is it possible in PDM to move this to a different group called "litestar-plugins" (not bikeshedded),that is still mandatory but just for cosmetic reasons so its easier to read?

File: litestar/contrib/htmx/_utils.py (hunk 1 above)
Comment (cofin) [reply litestar/contrib/htmx/_utils.py:8]: No, but I figured these methods may be used enough for us to have to have it re-exported.

File: tests/unit/test_template/test_template.py
Comment (provinzkraut) [inline tests/unit/test_template/test_template.py:1]: These changes seem unrelated?

File: tests/unit/test_testing/test_test_client.py
Comment (provinzkraut) [inline tests/unit/test_testing/test_test_client.py:1]: More unrelated changes?

**Review by provinzkraut — APPROVED** [review]

**Review by provinzkraut — CHANGES_REQUESTED** [review]
Right now this is a breaking change though, due to the fact that this line here sets the request class of the application to `HTMXRequest`.

https://github.com/litestar-org/litestar-htmx/blob/4f5fc871ee5011bcb4b3b422c476db09770bf574/litestar_htmx/plugin.py#L42

File: tests/unit/test_testing/test_test_client.py
Comment (cofin) [reply tests/unit/test_testing/test_test_client.py:1]: We have some issues after upgrading all of the latest dependencies.  

There's an issue in the latest version of AnyIO that makes the trio implementation fail on this test.

**Comment by cofin** [comment]
> Right now this is a breaking change though, due to the fact that this line here sets the request class of the application to `HTMXRequest`.
> 
> https://github.com/litestar-org/litestar-htmx/blob/4f5fc871ee5011bcb4b3b422c476db09770bf574/litestar_htmx/plugin.py#L42

There is no `HTMXPlugin` currently.  This is actually new functionality that I've implemented with the move from `contrib`.

File: tests/unit/test_template/test_template.py
Comment (cofin) [reply tests/unit/test_template/test_template.py:1]: This is the same as the issue below.  The tests no longer pass after upgrading the project dependencies.

**Review by provinzkraut — APPROVED** [review]
